docs/two-phase-commit-transactions.md

```diff
@@ -193,178 +215,88 @@ tx.commit();
-        }
-      } catch (CrudException | PreparationException | ValidationException | CommitException e) {
-        // If you catch CrudException or PreparationException or ValidationException or
-        // CommitException, it indicates some failure happens, so you should cancel the transaction
-        // or retry the transaction after the failure or error is fixed
-        try {
-          tx.rollback();
-        } catch (RollbackException ex) {
-          // Rolling back the transaction failed. You can log it here
-        }
-        return;
-      } catch (UnknownTransactionStatusException e) {
-        // If you catch `UnknownTransactionStatusException` when committing the transaction, you are
-        // not sure if the transaction succeeds or not. In such a case, you need to check if the
-        // transaction is committed successfully or not and retry it if it failed. How to identify a
-        // transaction status is delegated to users
-        return;
-      }
-    }
+TransactionFactory factory1 =
+    TransactionFactory.create("<configuration file path for the transaction manager1>");
+TwoPhaseCommitTransactionManager transactionManager1 =
+    factory1.getTwoPhaseCommitTransactionManager();
+
+TransactionFactory factory2 =
+    TransactionFactory.create("<configuration file path for the transaction manager2>");
+TwoPhaseCommitTransactionManager transactionManager2 =
+    factory2.getTwoPhaseCommitTransactionManager();
+
+TwoPhaseCommitTransaction transaction1 = null;
+TwoPhaseCommitTransaction transaction2 = null;
+try {
+  // Begin a transaction
+  transaction1 = transactionManager1.begin();
+
+  // Join the transaction begun by transactionManager1 with the transaction ID
+  transaction2 = transactionManager2.join(transaction1.getId());
+
+  // Execute CRUD operations in the transaction
+  Optional<Result> result = transaction1.get(...);
+  List<Result> results = transaction2.scan(...);
+  transaction1.put(...);
+  transaction2.delete(...);
+
+  // Prepare the transaction
+  transaction1.prepare();
+  transaction2.prepare();
+
+  // Validate the transaction
+  transaction1.validate();
+  transaction2.validate();
+
+  // Commit the transaction. If any of the transactions succeeds to commit, you can regard the
+  // transaction as committed
+  AtomicReference<Exception> exception = new AtomicReference<>();
+  boolean anyMatch =
+      Stream.of(transaction1, transaction2)
+          .anyMatch(
+              t -> {
+                try {
+                  t.commit();
+                  return true;
+                } catch (Exception e) {
+                  exception.set(e);
+                  return false;
+                }
+              });
+
+  // If all the transactions fail to commit, throw the exception and rollback the transaction
+  if (!anyMatch) {
+    throw exception.get();
+  }
+} catch (Exception e) {
+  // Rollback the transaction
+  if (transaction1 != null) {
+    transaction1.rollback();
```

Comment:
Wrapping this with try-catch would be safer just in case if `rollback()` could throw any exception.

Reply:
Fixed in 635b41d1ec91b4be3e98e835bd692a17a29407cb. Thanks.